src/twitchToDiscord.ts

```diff
@@ -0,0 +1,308 @@
+//Made by iSlammedMyKindle in 2022!
+//A bot that fuses the twitch chat and the desired discord channel together.
+
+//The bot isn't designed to scale for now. Using this to setup my personal server & twitch chat.
+import configFile from '../config.json';
+import tmijs from 'tmi.js';
+import fs from 'fs';
+import { authenticateTwitch } from './oauth';
+import { AnyChannel, Client, Collection, Message, PartialMessage, TextChannel } from 'discord.js';
+import { linkedListNode, nodeInterface } from './linkedList';
+import { conjoinedMsg, twitchMsg } from './messageObjects';
+
+const MAX_MSG_CACHE: number = 100; //Not a hard limit, it's more of a practical one
+let twitchClient: tmijs.Client;
+let currMsgCount: number = 0;
+let lastUserStateMsg: any;
+
+/**
+ * If an error happens either on twitch or discord, print the thing
+ * @param {*} error 
+ */
+const genericPromiseError = (error: any) => console.error('Snap, I hit a snag... >.<', error);
+
+//The jank method of finding twitch messages. If we get a message from discord, we send the message as the key here, then we use that to determine if the message is ours and then link it back to discord.
+const twitchMessageSearchCache: { [key: string]: linkedListNode; } = {},
+
+    //This is a linked list that will contain the last 100 messages.
+    messageLinkdListInterface = new nodeInterface(),
+
+    //This map will store discord & twitch messages, with the keys being the message objects. If we need to make edits, then we just go here to determine if the message is something we're storing already
+    discordTwitchCacheMap = new Map();
+
+function manageMsgCache(specificNode?: linkedListNode): number | linkedListNode
+{
+    if(!specificNode && currMsgCount < MAX_MSG_CACHE) return currMsgCount++;
+
+    //Delete messages once we hit our cache limit, or if we defined a node to delete, destroy that instead
+    if(!specificNode)
+        specificNode = messageLinkdListInterface.beginningNode as linkedListNode; //Garbage collection takes care of this, so need to run delete
+
+    messageLinkdListInterface.rebindForDelete(specificNode);
+
+    if(specificNode.data.twitchArray[0])
+        discordTwitchCacheMap.delete(specificNode.data.twitchArray[0]);
+
+    if(specificNode.data.discord)
+        discordTwitchCacheMap.delete(specificNode.data.discord);
+
+    return specificNode;
+}
+
+/**
+ * @description Deletes a twitch message
+ * @param {*} twitchObj I have no fucking clue as to what could be passed for this. It changes every other fucking call. 
```

Comment:
Entertaining, but my bias is professionalism over profanity. I'm gonna sound like dad, but I believe this should be exchanged with something else.